src/arch/ia32/interrupts/pic8259a.rs

```diff
@@ -0,0 +1,248 @@
+//! A module for the 8259 PIC.
+use crate::arch::io::port::Port;
+use crate::arch::io::register::WriteRegister;
+use crate::utils::bitfield::*;
+use log::trace;
+
+/// Master and Slave PICs ports on IBM PC
+const MASTER_PIC_COM: u16 = 0x20;
+const SLAVE_PIC_COM: u16 = 0xA0;
+
+/// PIC IRQs offset in IDT
+pub const PIC_OFFSET: usize = 32;
+// PIC IRQs mapping
+pub const PIT_IRQ: usize = 0;
+pub const KEYBOARD_IRQ: usize = 1;
+
+/// A structure representing the first initialisation command for a 8259 PIC.
+#[derive(Clone, Copy)]
+pub struct ICW1(u8);
+
+impl ICW1 {
+    /// Creates a [`ICW1`] byte from parameters.
+    ///
+    /// # Arguments
+    ///
+    /// * `has_icw4` - If true, ICW4 will be expected in initialisation phase.
+    /// * `single` - Single mode (true) or Cascaded mode (false).
+    /// * `adi` - CALL Address Inverval (true = 4, false = 8).
+    /// * `ltim` - Level Triggered mode (true) or Edge Triggered mode (false).
+    pub fn new(has_icw4: bool, single: bool, adi: bool, ltim: bool) -> Self {
+        Self(
+            0_u8.set_bit(0, has_icw4)
+                .set_bit(1, single)
+                .set_bit(2, adi)
+                .set_bit(3, ltim)
+                .set_bit(4, true),
+        )
+    }
+}
+
+impl From<ICW1> for u8 {
+    fn from(value: ICW1) -> Self {
+        value.0
+    }
+}
+
+/// Enum to describe the cascading role of the 8259 PIC.
+#[derive(PartialEq)]
+enum PicRole {
+    /// No cascading.
+    SINGLE,
```

Comment:
The case should be consistent with the other enums.

Reply:
Fixed in 1b7fa082f519efe83b62b109f000ef7bec0b1cda